pkg/bitcoin/electrum/electrum.go

```diff
@@ -679,7 +679,8 @@ func (c *Connection) verifyServer() error {
 	}
 
 	logger.Infof(
-		"connected to electrum server [version: [%s], protocol: [%s]]",
+		"connected to [%s] electrum server [version: [%s], protocol: [%s]]",
+		c.config.URL,
```

Comment:
I'm not sure we should print it in all the cases. What if the URL is configured by the user and contains authorization data? Maybe we could print the server in the function where we draw it when it's missing in the config?

Reply:
Good point. Unfortunately, printing the server in the function where we draw it gives a weird order of entries when starting the client:
```
2023-06-02T16:24:16.833+0200    INFO    keep-config     config/electrum.go:68   Auto-selecting Electrum server: [ssl://electrum.blockstream.info:50002]
2023-06-02T16:24:16.834+0200    INFO    keep-cmd        cmd/start.go:62 Starting the client against [mainnet] ethereum network...
```

We can do it given it is better to print it this way than not print at all but I'd like to explore the possibility of using a regex to remove possible credentials first.

Reply:
You may want to look at the url library to parse the string and cut off the credentials.

Reply:
I took a bit different approach, see https://github.com/keep-network/keep-core/pull/3596/commits/604f2a0355b272508b96b434189325a08de29ecb:
1. We only print Electrum server URL when it is auto-selected. This is consistent with the approach we use for Ethereum: we are not printing the URL to the logs because we expect the Ethereum server URL is set in the config. Printing the auto-selected URL is logged in the config logger because the auto-selection happens in the config.
2. We now print both Ethereum and Bitcoin networks used. This is also happening on the config level because those networks are resolved on the config level.

Reply:
Notes for future-us:

1. Redacting with `url` package is nice but it only redacts the password. I'd prefer to redact the username as well.
2. I do not think Fulcrum/ElectrumX allows passing username+password in the URL. This would be only secure on TLS-encrypted connections and I did not find anything about it in the config. If we ever see someone like Infura/Alchemy for Bitcoin, they will probably require to pass the project ID as a part of the path and allow only TLS-encrypted connections, just like Infura/Alchemy. In any case, we shouldn't output this URL to logs and that's why we are not outputting Ethereum server URL into logs.